### ISSUE TITLE
[class-literal-property-style] Literal property style does not respect inherited types

### ISSUE THREAD

**Issue by Paril** [typescript-eslint/typescript-eslint#3602]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/class-literal-property-style": "warn"
  }
}
```

```TS
abstract class BaseClass
{
    // class-literal-property-style error; this one is fair, because it isn't entirely aware
    // that I'm going to override it, so this one I can see just being an eslint disable
    get cursor(): string|string[] { return 'crosshair'; }

    readonly hidden: boolean = false;

    abstract get icon(): string;
}

class ChildClass extends BaseClass
{
    // class-literal-property-style error
    // this one, however, is interesting because it should know that it is
    // overriding an existing getter and can't be made a literal 
    override get cursor() { return 'square'; }

    readonly hidden: boolean = true

    readonly icon = 'circle'

    something: boolean
}
```

**Expected Result**

ESLint shouldn't suggest `class-literal-property-style` when overriding a getter, because it is not possible to turn an inherited getter into a readonly property (ts(2610))

**Actual Result**

It suggests breaking the code by attempting to convert the children into a literal when it is not possible. Conversely, it suggests converting the base class one into a literal, which will not work for children that need to use a getter for conditional return values (ts(2611)).

**Additional Info**

As a side-issue, this rule is not enum-aware and does not trigger if the literal returned from a getter is an enum value:

```ts
enum Test
{
    Value
}

class TestClass
{
    // should trigger here
    get test() { return Test.Value; }
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `8.2.1` |
| `@typescript-eslint/parser`        | `8.2.1` |
| `TypeScript`                       | `4.3.4` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `14.8.0` |

**Comment by bradzacher** [comment]
Why are you using a `get`ter in the first place for readonly properties instead of just using a readonly property?

```ts
class Foo {
    get value1(): string { return 'constant' } // bad
    readonly value2: string = 'constant'; // good
}

class Bar extends Foo {
	override get value1() { return 'other constant'; } // bad
	override readonly value2 = 'other constant'; // good
}
```

**Comment by Paril** [comment]
In my particular use-case, it's because implementors might use a getter that does not return a constant (for instance it may depend on other members' state at any given time).

It suggests `readonly` for the base class, which technically makes sense and could be ignored, but it shouldn't suggest it for child classes that can't legally be turned into a `readonly`.

**Comment by bradzacher** [comment]
> In my particular use-case, it's because implementors might use a getter that does not return a constant (for instance it may depend on other members' state at any given time).

That was the context that was missing from the original post!
With any issue it's best to explain *why* you want to do something a certain way.

---

I do agree that the rule should specifically ignore getters marked with the new `override` modifier, as that clearly marks them as "defined elsewhere".
